src/registrar/models/domain.py

```diff
@@ -492,11 +1058,17 @@ def _fetch_cache(self, fetch_hosts=False, fetch_contacts=False):
                     # we do not use _get_or_create_* because we expect the object we
                     # just asked the registry for still exists --
                     # if not, that's a problem
+                    if id is None:
```

Comment:
When implementing the get contact, here will need to factor that contacts will be a dictionary or object and id will need to be grabbed accordingly.